docs/src/examples/FOBP_example3.jl

```diff
@@ -0,0 +1,117 @@
+# # Foundations of Bilevel Programming: Example 3
+# This example is from the book _Foundations of Bilevel Programming_ by Stephan
+# Dempe, Chapter 3.7, Page 59. [url](https://www.springer.com/gp/book/9781402006319)
+
+
+# Model of the problem
+
+# First level
+# ```math
+# \min \sum_{i\in I} x_i - z,\\
+# \notag s.t.\\
+# y^a_i\geq 0, \forall i \in I,\\
+# y^a_i\leq 1, \forall i \in I,\\
+# y^b_i\geq 0, \forall i \in I,\\
+# y^b_i\leq 1, \forall i \in I,\\
+# y^a_i + y^b_i = 1, \forall i \in I,\\
+# z\geq 0,\\
+# z\leq 1,\\
+# y^a_1 + y^a_2 + y^a_3 \geq z,\\
+# -y^b_1 - y^b_4 + y^a_3 \geq z,\\
+# y^b_7 - y^b_6 + y^a_4 \geq z,\\
+# y^a_5 + y^a_6 + y^a_7 \geq z.\\
+# ```
+# Second level
+# ```math
+# \min -\sum_{i\in I}x_i,\\
+# \notag s.t.\\
+# \sum x_i \geq 0,\\
+# x_i \leq y^a_i, \forall i\in I,\\
+# x_i \leq y^b_i, \forall i\in I,\\
+# I = \{1,...,7\}
+# ```
+
+
+using BilevelJuMP
+using Ipopt
+using JuMP
+using Test
+
+model = BilevelModel(Ipopt.Optimizer, mode = BilevelJuMP.ProductMode(1e-9))
+
+# Global variables
+
+I = 7 # maximum literals
```

Comment:
Now's good time to read the Julia performance tips:
https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-global-variables

Reply:
Great, thanks. I assume you want to say don't use a global variable if you don't need to change in process and instead define constant variables.

Reply:
It's okay for this example.